### PR TITLE
SV/UVM coding style guidelines 

### PR DESCRIPTION
**Pull request by MikeOpenHWGroup**
SystemVerilog/UVM coding style guidelines from [Advanced UVM](https://www.amazon.ca/Advanced-UVM-Brian-Hunter-ebook/dp/B018UL698W) by Brian Hunter.  Cloned from https://github.com/bphunter1972/core-v-verif/blob/master/docs/CodingStyleGuidelines.md.

Note that this pull-request is subject to a CQ review.  I will remove the DO NOT MERGE label pending the review outcome.

### CODE REVIEW

**Comment by MikeOpenHWGroup** [comment]
[CQ 23796](http://dev.eclipse.org/ipzilla/show_bug.cgi?id=23796) for this pull-request has been completed and the submission is approved.

This PR is now ready for review by @DBees, and I will merge it in pending his approval.

**Review by DBees — APPROVED** [review]

**Pull request merged**
